test/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/GetRecordsControllerSpec.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.controllers
+
+import cats.data.EitherT
+import org.mockito.ArgumentMatchersSugar.any
+import org.mockito.MockitoSugar.when
+import org.scalatestplus.mockito.MockitoSugar
+import org.scalatestplus.play.PlaySpec
+import play.api.http.Status.{BAD_REQUEST, OK}
+import play.api.libs.json.Json
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{contentAsJson, defaultAwaitTimeout, status, stubControllerComponents}
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.eis.GoodsItemRecords
+import uk.gov.hmrc.tradergoodsprofilesrouter.models.response.errors.{Error, ErrorResponse, RouterError}
+import uk.gov.hmrc.tradergoodsprofilesrouter.service.{RouterService, UuidService}
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.{ApplicationConstants, HeaderNames}
+
+import scala.concurrent.ExecutionContext
+import scala.language.postfixOps
+
+class GetRecordsControllerSpec extends PlaySpec with MockitoSugar {
+
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  val mockRouterService = mock[RouterService]
+  val mockUuidService   = mock[UuidService]
+
+  private val sut =
+    new GetRecordsController(
+      stubControllerComponents(),
+      mockRouterService,
+      mockUuidService
+    )
+
+  def validHeaders: Seq[(String, String)] = Seq(
+    HeaderNames.ClientId -> "clientId"
+  )
+
+  "GET /:eori/record/:recordId" should {
+
+    "return a successful JSON response for a single record" in {
+
+      when(mockRouterService.fetchRecord(any, any)(any, any))
+        .thenReturn(EitherT.rightT(getSingleRecordResponseData))
+
+      val result = sut.getTGPRecord("GB123456789001", "12345")(
+        FakeRequest().withHeaders(validHeaders: _*)
+      )
+      status(result) mustBe OK
+      withClue("should return json response") {
+        contentAsJson(result) mustBe Json.toJson(getSingleRecordResponseData)
+      }
+    }
+    "return 400 Bad request when mandatory request parameter EORI is missing" in {
```

Comment:
I would not worry about this test, so delete it, as this will be take care by the network layer. So really here we are testing Play framework or the network layer which we should not do.

Also I woiuld add a test case when the EISConnecotor fail to retrieve the record